gwcs/api.py

```diff
@@ -172,31 +173,54 @@ def world_axis_object_components(self):
         raise NotImplementedError()
 
     # High level APE 14 API
-
     def low_level_wcs(self):
         """
         Returns a reference to the underlying low-level WCS object.
         """
         return self
 
+    def _sanitize_pixel_inputs(self, *pixel_arrays):
+        pixels = []
+        if self.forward_transform.uses_quantity:
+            for i, pixel in enumerate(pixel_arrays):
+                if not isinstance(pixel, u.Quantity):
+                    pixel = pixel * self.input_frame.unit[i]
+                pixels.append(pixel)
+        else:
+            for pixel in pixel_arrays:
+                if isinstance(pixel, u.Quantity):
+                    if pixel.unit != u.pixel:
```

Comment:
Sorry to come back to this but isn't this assuming the input is always in pixel units? Should it instead compare with the unit in `input_frame`? I think the APE calls the inputs `pixel_arrays`  but does not assume they are in pixels. Maybe. I hope so anyway.

Reply:
Fixed, thanks!